app/models/rule.rb

```diff
@@ -45,6 +45,14 @@ class Rule < ApplicationRecord
     where(benchmark_id: ::Xccdf::Benchmark.latest.pluck(:id))
   }
 
+  scope :canonical, lambda {
+    includes(:profiles).where(profiles: { id: Profile.canonical })
+  }
+
+  def canonical?
+    (profiles & Profile.canonical).any?
+  end
```

Comment:
Convenience methods. A rule is canonical if it is a member of any canonical profile.